opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -407,23 +407,23 @@ public static void addObservation(Event e, JSONObject jsonObject) {
         if (StringUtils.isBlank(value)) { return; }
 
         String type = getString(jsonObject, AllConstants.TYPE);
-        String entity =  getString(jsonObject, OPENMRS_ENTITY);
+        String entity =  CONCEPT;
         if (AllConstants.CHECK_BOX.equals(type)) {
             try {
                 List<Object> optionValues = new ArrayList<>();
                 if (jsonObject.has(AllConstants.OPTIONS)) {
                     JSONArray options = jsonObject.getJSONArray(AllConstants.OPTIONS);
+                    String checkboxEntity = getString(jsonObject, OPENMRS_ENTITY);
                     for (int i = 0; i < options.length(); i++) {
                         JSONObject option = options.getJSONObject(i);
                         boolean optionValue = option.optBoolean(VALUE);
-                        entity = getString(jsonObject, OPENMRS_ENTITY);
```

Comment:
I think this should be returned as
`entity = jsonObject.optString(OPENMRS_ENTITY,checkboxEntity);`

This is because options can have `OPENMRS_ENTITY` i.e concepts defined in the options